common/src/main/java/io/homeassistant/companion/android/common/bluetooth/ble/TransmitterManager.kt

```diff
@@ -58,9 +63,20 @@ object TransmitterManager {
         }
         val bluetoothAdapter = context.getSystemService<BluetoothManager>()?.adapter
         val bluetoothOn = bluetoothAdapter?.isEnabled == true
+        beaconManager = BeaconManager.getInstanceForApplication(context)
+
         if (bluetoothOn) {
             val beacon = buildBeacon(haTransmitter)
             if (!physicalTransmitter.isStarted) {
+                val builder = beaconNotification(true, context)
+                beaconManager?.enableForegroundServiceScanning(builder.build(), 445)
+                beaconManager?.setEnableScheduledScanJobs(false)
+                beaconManager?.beaconParsers?.clear()
+                beaconManager?.backgroundBetweenScanPeriod = Long.MAX_VALUE
+                beaconManager?.backgroundScanPeriod = 0
+                beaconManager?.foregroundBetweenScanPeriod = Long.MAX_VALUE
+                beaconManager?.foregroundScanPeriod = 0
+                beaconManager?.startMonitoring(region)
```

Comment:
Might be good to add a comment here explaining this enables a foreground service to improve reliability, so that one doesn't need to blame the code to find out why we are scanning in a monitoring service.

Minor cleanup so you don't need `beaconManager?.` everywhere:

```kotlin
beaconManager?.apply {
    enableForegroundServiceScanning(builder.build(), 445)
    setEnableScheduledScanJobs(false)
    // etc
}
```